react/src/utils/measureRequestDuration.js

```diff
@@ -0,0 +1,22 @@
+
+import * as Sentry from '@sentry/react';
+
+/**
+ * Measure the duration of a request and send it to Sentry as a custom metric
+ * @param {string} endpoint the endpoint that was called
+ * @returns {() => void} a function to stop the measurement
+ */
+export default function measureRequestDuration(endpoint) {
+  const start = Date.now();
+  
+  function stopMeasurement() {
+    const end = Date.now();
+    const duration = end - start;
+    Sentry.metrics.distribution('request.duration', duration, {
+      unit: 'milliseconds',
```

Comment:
@ArthurKnaus The `request.duration` metric does not seem to show any "samples" on the table below ([here](https://demo.sentry.io/ddm/?project=5808623&statsPeriod=7d&widgets=%5B%7B%22mri%22%3A%22d%3Acustom%2Frequest.duration%40milliseconds%22%2C%22op%22%3A%22sum%22%2C%22query%22%3A%22%22%2C%22displayType%22%3A%22line%22%2C%22showSummaryTable%22%3Atrue%2C%22sort%22%3A%7B%22order%22%3A%22asc%22%7D%7D%5D))
Do you know why this is happening? I noticed that here the unit is set to
![Screenshot 2024-01-22 at 12 17 27 PM](https://github.com/sentry-demos/empower/assets/109162568/14aa6795-7ba5-470b-a5ea-f5942827b11a)
 `miliseconds` while in the UI, the other distribution metrics have `milisecond` as the unit

Reply:
@serglom21 good catch! the unit should be `millisecond`, updating...
Regarding the samples:
For now the JS SDK does not support sample correlation. So this is to be expected.